packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -149,6 +149,17 @@ export interface FunctionAttributes {
    * to this Lambda.
    */
   readonly securityGroup?: ec2.ISecurityGroup;
+
+  /**
+   * Setting this property informs the CDK that permissions can be added to this function.
+   * When not configured, the CDK attempts to auto-determine this. If determined as false, adding permission
+   * to this function will fail.
+   *
+   * Set this to property *ONLY IF* the imported function is in the same account as the stack
+   * it's imported in.
+   * @default - true, if the Stack is configured with an `env` and the account is the same as this function. false, otherwise.
+   */
+  readonly allowPermissions?: boolean;
```

Comment:
`~` `allowPermissions` doesn't quite read right to me. Maybe `allowCreatingPermissions` or `canCreatePermissions` (like the internal field)?

Reply:
I agree that the name is not ideal.

What is the user intent here? Is the intent to block/forbid or to enable some behavior despite the default?

Reply:
It is to enable a behaviour despite the default. Any suggestions?

Reply:
Specifically to enable the ability to modify permissions of an imported function.

Reply:
Modify permissions for an imported function in environment agnostic stacks?

Would it make sense that this will be named something like `assumeSameEnvironment`?

Reply:
I don't mind calling it that. This is going to be a rarely used property with no way to easily describe it via a name, so I'm not fussed.

Reply:
I think `sameEnvironment` is actually better because the default is conditional